src/nvim/regexp_defs.h

```diff
@@ -41,18 +43,38 @@
 #define NFA_ENGINE          2
 
 typedef struct regengine regengine_T;
+typedef struct regprog regprog_T;
+typedef struct reg_extmatch reg_extmatch_T;
```

Comment:
Might as well give this treatment to regmmatch_T too.

Also the linter will want the `/*`-style comments changed to `//`-style.

Reply:
@relnod Is this ordering still needed, now that buffer_defs.h includes the new regexp_2_defs.h ?

I see the last commit is marked WIP. Do you want to finish it, or maybe break it into a new PR, so that we can merge the previous changes?